otherlibs/systhreads/st_posix.h

```diff
@@ -356,19 +356,20 @@ static void st_decode_sigset(value vset, sigset_t * set)
 
 static value st_encode_sigset(sigset_t * set)
 {
-  value res = Val_int(0);
+  CAMLparam0();
+  CAMLlocal1(res);
   int i;
 
-  Begin_root(res)
-    for (i = 1; i < NSIG; i++)
-      if (sigismember(set, i) > 0) {
-        value newcons = caml_alloc_small(2, 0);
-        Field(newcons, 0) = Val_int(caml_rev_convert_signal_number(i));
-        Field(newcons, 1) = res;
-        res = newcons;
-      }
-  End_roots();
-  return res;
+  res = Val_emptylist;
+
+  for (i = 1; i < NSIG; i++)
+    if (sigismember(set, i) > 0) {
+      value newcons = caml_alloc_small(2, 0);
+      Field(newcons, 0) = Val_int(caml_rev_convert_signal_number(i));
+      Field(newcons, 1) = res;
```

Comment:
cf. otherlibs/unix/signals.c, while here:
```suggestion
      value newcons = caml_alloc_2(0,
        Val_int(caml_rev_convert_signal_number(i)),
        res);
```

Reply:
Ack. Is it safe to replace `newcons` by `res` and getting rid of `res = newcons` in line 370?

Reply:
Fixed.